acvm-repo/brillig_vm/src/arithmetic.rs

```diff
@@ -118,67 +197,79 @@ fn evaluate_binary_int_op_u1(
     rhs: bool,
 ) -> Result<bool, BrilligArithmeticError> {
     let result = match op {
-        BinaryIntOp::Add | BinaryIntOp::Sub => lhs ^ rhs,
-        BinaryIntOp::Mul => lhs & rhs,
+        BinaryIntOp::Equals => lhs == rhs,
+        BinaryIntOp::LessThan => !lhs & rhs,
+        BinaryIntOp::LessThanEquals => lhs <= rhs,
+        BinaryIntOp::And | BinaryIntOp::Mul => lhs & rhs,
+        BinaryIntOp::Or => lhs | rhs,
+        BinaryIntOp::Xor | BinaryIntOp::Add | BinaryIntOp::Sub => lhs ^ rhs,
         BinaryIntOp::Div => {
             if !rhs {
                 return Err(BrilligArithmeticError::DivisionByZero);
             } else {
                 lhs
             }
         }
+        _ => unreachable!("Operators not handled by this function"),
+    };
+    Ok(result)
+}
+
+fn evaluate_binary_int_op_cmp<T: Ord + PartialEq>(op: &BinaryIntOp, lhs: T, rhs: T) -> bool {
+    match op {
         BinaryIntOp::Equals => lhs == rhs,
-        BinaryIntOp::LessThan => !lhs & rhs,
+        BinaryIntOp::LessThan => lhs < rhs,
         BinaryIntOp::LessThanEquals => lhs <= rhs,
-        BinaryIntOp::And => lhs & rhs,
-        BinaryIntOp::Or => lhs | rhs,
-        BinaryIntOp::Xor => lhs ^ rhs,
-        BinaryIntOp::Shl | BinaryIntOp::Shr => {
-            if rhs {
-                false
+        _ => unreachable!("Operators not handled by this function"),
+    }
+}
+
+fn evaluate_binary_int_op_shifts<
+    T: AsPrimitive<usize> + From<u8> + Zero + PartialOrd + Shl<Output = T> + Shr<Output = T>,
```

Comment:
Is `T: AsPrimitive<usize>` required? It looks like only `rhs.as_()` is called, and that's already a `u8`.

Reply:
Ah, good spot. This is some leftover code from when both arguments were type `T`. I've restricted the type requirements to the minimum necessary.